packages/jest-runtime/src/transform.js

```diff
@@ -245,13 +245,21 @@ const stripShebang = content => {
   }
 };
 
-const instrumentFile = (content: string, filename: Path): string => {
+const instrumentFile = (
+  content: string,
+  filename: Path,
+  config: Config,
+  hasBeenPreprocessed: boolean
+): string => {
   // NOTE: Keeping these requires inside this function reduces a single run
   // time by 2sec if not running in `--coverage` mode
   const babel = require('babel-core');
   const babelPluginIstanbul = require('babel-plugin-istanbul').default;
 
-  if (babel.util.canCompile(filename)) {
+
+  // if the file has already been preprocessed, that means `content` contains
+  // valid javascript (even if the extention is '.ts', '.coffee'...)
+  if (hasBeenPreprocessed || babel.util.canCompile(filename)) {
```

Comment:
wait, why do we need this check at all? Shouldn't babel *always* be able to compile the source at this point?